common/src/main/java/org/apache/sedona/common/FunctionsGeoTools.java

```diff
@@ -134,4 +136,31 @@ public static Geometry voronoiPolygons(Geometry geom, double tolerance, Geometry
         }
         return builder.getDiagram(FunctionsGeoTools.GEOMETRY_FACTORY);
     }
+
+    public static Geometry bufferSpheroid(Geometry geometry, double radius, BufferParameters params){
+        // Determine the best SRID for spheroidal calculations
+        int bestCRS = Functions.bestSRID(geometry);
```

Comment:
Why not first check if the originalCRS is in WGS84 or not? If not, there is no need to call `BestSRID`. To make things neat, I recommend that we implement some sanity checks:

Sanity check Step 1 of `BestSRID`:

If the SRID is not given, we assume it is 4326.
If the SRID is given but is not 4326, we throw an exception saying that this function requires WGS84 / EPSG:4326 coordinates, and your SRID is XXX.

Sanity check Step 2 of `BestSRID`:

Given a geometry of 4326, its X must be in [-180, 180], its Y must be in [-90, 90]. If not, we throw an exception saying that this function requires WGS84 / EPSG:4326 coordinates but the bounding box of your geometry is XXXX.

Reply:
For Check 1 - 

> If the SRID is not given, we assume it is 4326.

This happens at [line 147](https://github.com/prantogg/sedona/blob/977ee844a87d589785feb8801252622aca278160/common/src/main/java/org/apache/sedona/common/FunctionsGeoTools.java#L147).


> If the SRID is given but is not 4326, we throw an exception saying that this function requires WGS84 / EPSG:4326 coordinates, and your SRID is XXX.

Having this exception would mean we disallow bestSRID to be determined for other Lon/lat coordinates systems such as NAD83 / EPSG:4269, ED50 (European Datum 1950) / EPSG:4230 and GDA94 / EPSG:4283

Check 2 can definitely be useful.

Reply:
@prantogg I thought Check 1 already disallowed NAD83 / EPSG:4269, ED50 (European Datum 1950) / EPSG:4230 and GDA94 / EPSG:4283?

Is the bestSRID supposed to be used for NAD83 / EPSG:4269, ED50 (European Datum 1950) / EPSG:4230 and GDA94 / EPSG:4283?

Reply:
Users that are sure about the bestSRID for their geometries can skip ST_Buffer(useSpheroid=true) altogether by manually utilizing ST_transform with ST_Buffer like so,
```
ST_Transform(ST_Buffer(ST_Transform(geometry, <original SRID>, <desired best SRID>), radius), <desired best SRID>, <original SRID>)
```

Maybe we could point this out in the documentation?